### PR TITLE
Diffsynth

### PR DESCRIPTION
**Pull request by akiseakusa**
Draft PR Description:

This PR (#12221) is a draft aimed at adding support for DiffSynth Blockwise ControlNet in diffusers.
References / Useful Links:

Model implementation & weights: [Canny ControlNet](https://huggingface.co/SahilCarterr/Qwen-Image-Blockwise-ControlNet-Canny)

Model implementation & weights: [Depth ControlNet](https://huggingface.co/SahilCarterr/Qwen-Image-Blockwise-ControlNet-Depth)

The PR is open for feedback, testing, and further adjustments from the community.

### CODE REVIEW

File: src/diffusers/pipelines/qwenimage/__init__.py [hunk 1]
```diff
@@ -25,7 +25,11 @@
     _import_structure["modeling_qwenimage"] = ["ReduxImageEncoder"]
     _import_structure["pipeline_qwenimage"] = ["QwenImagePipeline"]
     _import_structure["pipeline_qwenimage_controlnet"] = ["QwenImageControlNetPipeline"]
+<<<<<<< Updated upstream
```
Comment (yiyixuxu) [inline src/diffusers/pipelines/qwenimage/__init__.py:28]: can you remove these?

File: src/diffusers/pipelines/qwenimage/__init__.py [hunk 2]
```diff
@@ -25,7 +25,11 @@
...
     _import_structure["pipeline_qwenimage_controlnet_inpaint"] = ["QwenImageControlNetInpaintPipeline"]
+=======
```
Comment (yiyixuxu) [inline src/diffusers/pipelines/qwenimage/__init__.py:30]:
```suggestion
```

File: src/diffusers/pipelines/qwenimage/__init__.py [hunk 3]
```diff
@@ -25,7 +25,11 @@
...
+    _import_structure["pipeline_qwenimage_block_controlnet"] = ["QwenImageBlockControlNetPipeline"]
+>>>>>>> Stashed changes
```
Comment (yiyixuxu) [inline src/diffusers/pipelines/qwenimage/__init__.py:32]:
```suggestion
```

File: src/diffusers/pipelines/qwenimage/pipeline_qwenimage_blockcontrolnet.py [hunk 4]
```diff
@@ -0,0 +1,904 @@
...
+
+        if do_classifier_free_guidance and not guess_mode:
+            image = torch.cat([image] * 2)
+
+        return image
+
+    
+    @torch.no_grad()
+    @replace_example_docstring(EXAMPLE_DOC_STRING)
+    def __call__(
```
Comment (yiyixuxu) [inline src/diffusers/pipelines/qwenimage/pipeline_qwenimage_blockcontrolnet.py:514]: pipeline looks like it's same as instantx one, no?
https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/qwenimage/pipeline_qwenimage_controlnet.py

File: src/diffusers/models/autoencoders/autoencoder_kl_qwenimage.py [hunk 5]
```diff
@@ -1068,3 +1068,4 @@ def forward(
             z = posterior.mode()
         dec = self.decode(z, return_dict=return_dict)
         return dec
+
```
Comment (yiyixuxu) [inline src/diffusers/models/autoencoders/autoencoder_kl_qwenimage.py:1071]:
```suggestion
```

File: src/diffusers/models/controlnets/controlnet_qwenimage_blockwise.py [hunk 6]
```diff
@@ -0,0 +1,409 @@
...
+
+logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
+
+
+@dataclass
+class QwenImageBlockControlNetOutput(BaseOutput):
+    controlnet_block_samples: Tuple[torch.Tensor]
+
+
+class BlockWiseControlBlock(torch.nn.Module):
```
Comment (yiyixuxu) [inline src/diffusers/models/controlnets/controlnet_qwenimage_blockwise.py:45]: is this the only difference between Diffsynth & instantX controlnet, can we just add the change to the existing one?

File: src/diffusers/pipelines/qwenimage/pipeline_qwenimage_blockcontrolnet.py (hunk 4 above)
Comment (akiseakusa) [reply src/diffusers/pipelines/qwenimage/pipeline_qwenimage_blockcontrolnet.py:514]: It does seem same at first, but controlnet is different, with some parameter and scheduler changes.
